File: 383-ransom-note/ransom-note.py
class Solution:
    def canConstruct(self, ransomNote: str, magazine: str) -> bool:
        if len(ransomNote) > len(magazine): return False
        vault = list(magazine)
        for c in ransomNote:
            if c not in vault:
                return False
            vault.remove(c)
        return True

File: 383-ransom-note/ransom-note.cpp
class Solution {
public:
    bool canConstruct(string ransomNote, string magazine) {
        if (ransomNote.size() > magazine.size()) {
            return false;
        }
        std::unordered_map<char, int> vault;
        for (char c : magazine) {
            vault[c] += 1;
        }
        for (char c : ransomNote) {
            vault[c] -= 1;
            if (vault[c] < 0) {
                return false;
            }
        }
        return true;
    }
};